library/src/main/java/com/alamkanak/weekview/WeekView.kt

```diff
@@ -0,0 +1,2525 @@
+     * The zoom will always be focused at the center of your gesture.
+     *
+     * @param zoomFocusPointEnabled whether the zoomFocusPoint is enabled
+     */
+    var isZoomFocusPointEnabled = true
+    /**
+     * Get scroll duration
+     *
+     * @return scroll duration
+     */
+    /**
+     * Set the scroll duration
+     *
+     * @param scrollDuration the new scrollDuraction
+     */
+    var scrollDuration = 250
+    var timeColumnResolution = 60
+    var typeface: Typeface? = Typeface.DEFAULT_BOLD
+        set(value) {
+            if (value != null) {
+                field = value
+                init()
+            }
+        }
+
+    private var mMinTime = 0
+    private var mMaxTime = 24
+
+    /**
+     * auto calculate limit time on events in visible days.
+     */
+    var autoLimitTime = false
+        set(value) {
+            field = value
+            invalidate()
+        }
+
+    var isDropListenerEnabled = false
+        set(value) {
+            field = value
+            setOnDragListener(if (value) DragListener() else null)
+        }
+    var minOverlappingMinutes = 0
+    private var mIsScrollNumberOfVisibleDays = false
+
+    // Listeners.
+    var eventClickListener: EventClickListener? = null
+    var eventLongPressListener: EventLongPressListener? = null
+    /**
+     * Get event loader in the week view. Event loaders define the  interval after which the events
+     * are loaded in week view. For a MonthLoader events are loaded for every month. You can define
+     * your custom event loader by extending WeekViewLoader.
+     *
+     * @return The event loader.
+     */
+    /**
+     * Set event loader in the week view. For example, a MonthLoader. Event loaders define the
+     * interval after which the events are loaded in week view. For a MonthLoader events are loaded
+     * for every month. You can define your custom event loader by extending WeekViewLoader.
+     *
+     * @param loader The event loader.
+     */
+    var weekViewLoader: WeekViewLoader? = null
+    var emptyViewClickListener: EmptyViewClickListener? = null
+    var emptyViewLongPressListener: EmptyViewLongPressListener? = null
+    private var mDateTimeInterpreter: DateTimeInterpreter? = null
+    var scrollListener: ScrollListener? = null
+    var addEventClickListener: AddEventClickListener? = null
+    var weekViewDropListener: DropListener? = null
+    var enableDrawHeaderBackgroundOnlyOnWeekDays = false
+        set(value) {
+            field = value
+            invalidate()
+        }
+    var sideTitleText: String? = null
+        set(value) {
+            field = value
+            invalidate()
+        }
+    private val sideTitleTextPaint = Paint(Paint.ANTI_ALIAS_FLAG)
+
+    private val mGestureListener = object : GestureDetector.SimpleOnGestureListener() {
+
+        override fun onDown(e: MotionEvent): Boolean {
+            mStartOriginForScroll = mCurrentOrigin.x
```

Comment:
Delete the line `mStartOriginForScroll = mCurrentOrigin.x` This don't need to be done at each onDown.